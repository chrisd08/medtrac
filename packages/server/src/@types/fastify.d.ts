import * as http from "http";

declare module "fastify" {
  export interface FastifyInstance<
    HttpServer = http.Server,
    HttpRequest = http.IncomingMessage,
    HttpResponse = http.ServerResponse
  > {
    authenticate: FastifyMiddleware;
  }

  interface FastifyRequest<
    HttpRequest = http.IncomingMessage,
    Query = DefaultQuery,
    Params = DefaultParams,
    Headers = DefaultHeaders,
    Body = DefaultBody
  > {
    user: unknown;
  }
}
