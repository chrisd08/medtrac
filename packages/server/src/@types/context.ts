import { UserAPI } from "../api";

interface API {
  user?: UserAPI;
}

export interface Context {
  dataSources?: API;
  url: string;
  req: Express.Request;
  user: {
    sub: string;
  };
}
