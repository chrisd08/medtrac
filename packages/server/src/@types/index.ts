export * from "./context";
export * from "./fastify.d";
export * from "./repos";
