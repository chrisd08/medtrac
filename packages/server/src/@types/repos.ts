import { Repository } from "typeorm";
import { Profile, User } from "../entity";

export interface DataSourceRepos {
  userRepo?: Partial<Repository<User>>;
  profileRepo?: Partial<Repository<Profile>>;
}
