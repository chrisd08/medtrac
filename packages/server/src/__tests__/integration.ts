import { ApolloServerTestClient } from "apollo-server-testing";
import gql from "graphql-tag";
import { Connection } from "typeorm";
import { DataSourceRepos } from "../@types";
import { setupTests } from "./__utils";

const ME_QUERY = gql`
  {
    me {
      id
      username
      profiles {
        name
      }
    }
  }
`;

let connection: Connection,
  client: ApolloServerTestClient,
  repos: DataSourceRepos;

beforeAll(async () => {
  [connection, client, repos] = await setupTests();
});

afterAll(() => connection.close());

describe("user", () => {
  describe("when a user does not exist", () => {
    it("should create one with a default profile", async () => {
      const { query } = client;
      const response = await query({ query: ME_QUERY });
      expect(response.data.me).toMatchSnapshot({ id: expect.any(String) });
    });
  });

  describe("when a user does exist", () => {
    beforeAll(() => {
      repos.userRepo.create({ username: "test" });
    });

    afterAll(() => {
      repos.userRepo.delete({ username: "test" });
    });

    it("returns the user", async () => {
      const { query } = client;
      const response = await query({ query: ME_QUERY });
      expect(response.data.me).toMatchSnapshot({ id: expect.any(String) });
    });
  });
});
