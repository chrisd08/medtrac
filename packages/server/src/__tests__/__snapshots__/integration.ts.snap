// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`user when a user does exist returns the user 1`] = `
Object {
  "id": Any<String>,
  "profiles": Array [
    Object {
      "name": "default",
    },
  ],
  "username": "test",
}
`;

exports[`user when a user does not exist should create one with a default profile 1`] = `
Object {
  "id": Any<String>,
  "profiles": Array [
    Object {
      "name": "default",
    },
  ],
  "username": "test",
}
`;
